### ISSUE TITLE
How long to train?

### ISSUE THREAD

**Issue by PeterHickman** [domesticatedviking/TextyMcSpeechy#5]
Sorry that this is not a real issue but I was wondering how long I would need to run for in epochs before there is a usable model. Presently it is taking me 22 minutes per epoch (rtx3070) and after 70 epochs there is not the slightest hint of the target voice

Should I be looking at 100s, 1,000s or 10,000s epochs?

**Comment by domesticatedviking** [comment]
With the voices I've trained the target voice starts to appear within a few hundred epochs and are usually heading into overtrained territory between 1000-2000 epochs.   The time per epoch varies pretty significantly with the size of the dataset.    How many files are you training on?

**Comment by PeterHickman** [comment]
I downloaded a bunch of Youtube videos of Patrick Boyle (about 44 were usable) and used the subtitles to extract 18,292 wav files and am using that. I sort of assumed that you would need a lot of samples. Have I over done it?

**Comment by domesticatedviking** [comment]
That sounds very much like overkill for finetuning a pretrained tts model, but now you've got me quite curious about the quality you'll ultimately get out of a dataset that large.   The largest dataset I've used for training had ~400 samples, and I've gotten very good results for some of my character voices with only 50 samples, especially when the character's catchphrases were in the dataset.

**Comment by domesticatedviking** [comment]
By the way, not sure if those 18000 wav files contained individual words, but if they did, you will want to find a way to process them as sentences, otherwise your intonation will be very robotic.    A good dataset for training TTS will include phrases of various lengths.

**Comment by PeterHickman** [comment]
Trying with a smaller dataset, 500 samples, and I'm getting an epoch a
minute at the moment. Not sure that I will live long enough for my original
dataset to run for 1,000 epochs (+900 days at 22 minutes an epoch)

**Comment by PeterHickman** [comment]
The samples are things like:

9daKf7AdJp8_00195|0|Another great wave of international trade came with the
Industrial Revolution when new
9daKf7AdJp8_00196|0|technologies like the steam engine meant that goods
could be transported over thousands
9daKf7AdJp8_00197|0|of miles, both within countries and across borders.
9daKf7AdJp8_00198|0|Britain, being the source of so much of the capital and
technology of the day, benefited
9daKf7AdJp8_00199|0|the most from international trade at this time.
9daKf7AdJp8_00200|0|John Maynard Keynes wrote in his book “The Economic
Consequences of the Peace” about
9daKf7AdJp8_00201|0|how lifestyles in Britain improved at this time when,
conveniences, comforts and amenities
9daKf7AdJp8_00202|0|became available to the middle classes that had been
beyond the reach of the richest and
9daKf7AdJp8_00203|0|most powerful monarchs of other ages.
9daKf7AdJp8_00204|0|He wrote “The inhabitant of London could order by
telephone, sipping his morning tea

**Comment by PeterHickman** [comment]
The base checkpoint I used was at epoch 9029. Overnight I trained to epoch 10157 and then ran a test sample and it clearly sounds like the target voice (Patrick Boyle). I will probably train a 1,000 epochs more to see if that makes any difference and also try with fewer samples to get a feel for how much is enough

But overall I have proven to myself that Youtube videos where there is a single speaker and a transcript is available is a usable source of training data for deepfaking voices of Youtube creators (probably a very niche area of interest)

At some point I will write this up and post it to Github and Reddit with a link to your project

Thanks for your help and this project too. I will close and sign off for now
